upgrade rubocop, add rubocop-minitest and correct offenses

dorianmariefr opened the pull request:
Closes #2850

sonalkr132 commented on Gemfile:
```diff
@@ -1,6 +1,6 @@
 source "https://rubygems.org"
 
-gem "rails", "~> 6.1.0"
+gem "rails"
```
please don't remove these version requirements. all of them are there because we need to carefully test our code before updating them.
you are welcome to update rubocop and related things but we can't make bulk changes to dependencies like this.

dorianmariefr replied:
sure, I added the versions back 7d2fd5c5e40c50ac6776bdebae59bc71de5b03d9

I still think we should trust the tests and upgrade dependencies

hsbt commented:
It's good to add `rubocop-minitest` and formatting the current codebase. But this PR contains a lot of other changes on lockfile.

@dorianmariefr Can you update the only rubocop related gems?

dorianmariefr commented:
@hsbt done, I reset the Gemfile.lock to master's version and did `bundle update rubocop rubocop-minitest rubocop-performance rubocop-rails`

sonalkr132 commented on Gemfile.lock:
```diff
@@ -477,4 +480,4 @@ DEPENDENCIES
 
 BUNDLED WITH
-   2.2.22
+   2.2.31
```
please don't change this. if you like us to update bundler version, please [update rubygems_version instead](https://github.com/rubygems/rubygems.org/commit/b0a413fa55f575aa6df2900175b726caf8a71e9b) in a separate PR

dorianmariefr replied:
or made a PR https://github.com/rubygems/rubygems.org/pull/2869 and reverted the change d717012e

sonalkr132 commented on lib/gemcutter/middleware/hostess.rb:
```diff
@@ -1,51 +1,52 @@
 module Gemcutter::Middleware
-  class Hostess < Rack::Static
```
I would suggest not to change this. I understand the reasoning of https://rubystyle.guide#namespace-definition, however, I don't think these classes are offenders and rarely see any change.
these classes have created issues with rails autoloader in the past.

dorianmariefr replied:
oh that's why I was getting loading errors, will disable this rule

dorianmariefr replied:
fixed in c7d05c845b7e903d27f12f918ae9263447f3b492

ashmaroli commented:
@dorianmariefr @hsbt Could one of you add the following line (w/o the backticks) to the end of the opening post?

```
Closes #2850
```

ashmaroli commented:
Closes #2850

dorianmariefr commented:
I'm renamed the PR, added the "closes" in the description and the commit, and rebased my commits to be one commit (I don't know if you merge or squash)

simi commented on Gemfile:
```diff
@@ -58,9 +58,10 @@ group :development, :test do
-  gem "rubocop", "0.93.1", require: false
-  gem "rubocop-rails", "2.9.1", require: false
-  gem "rubocop-performance", "1.10.2", require: false
+  gem "rubocop", require: false
```
Any reason to unlock those at all?

dorianmariefr replied:
so that we don't get stuck on a specific version and we follow the rubocop versions (e.g. not getting too much behind) ?

dorianmariefr replied:
if there is a new cop we don't want, we can disable it, or usually autocorrect it

simi replied:
Then please add at least some constraints if possible, not locked to individual versions, but opened using `~` on minor changes for example.

dorianmariefr replied:
I don't really agree, but sure I added some constraints on minor versions

dorianmariefr commented:
I think it was a bad idea to rebase, the PR was approved and now I guess it need to be approved again

simi commented:
Anyone any idea on `codeclimate/diff-coverage` CI fail?

sonalkr132 commented:
@dorianmariefr thank you for working on this.

> Anyone any idea on codeclimate/diff-coverage CI fail?

files changed were already not covered (we don't have 100% coverage).

dorianmariefr commented:
Thanks you for merging 🤩
